server/data/data-api/src/main/java/org/apache/james/user/api/UsersRepository.java

```diff
@@ -142,4 +142,10 @@
      */
     boolean isAdministrator(String username) throws UsersRepositoryException;
 
+    /**
+     * @return true if one can use {@link UsersRepository#updateUser(User)} {@link UsersRepository#addUser(String, String)}
```

Comment:
True, we would have Guice injection problem before.

Thus we need to design:
 - Filter injection, with their context. Maybe even in their own Guice module.
 - Split everything that can create a user (WebAdmin easy, CLI is hard as it's not modular)
 - We would need to combine modules in Cassandra-LDAP in another way that just override modules.

Thus, it brings complexity. If we are willing to pay it, then let it be. I'll close this PR, and update the ticket about the implementation requirements (seems hudge).